azure/packages/external-controller/src/AzureFunctionTokenProvider.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import axios from "axios";
+
+import { ITokenProvider, ITokenResponse } from "@fluidframework/routerlicious-driver";
+
+interface AzureMember<T = any> {
+	userName: string;
+	additionalDetails?: T;
+	userId: string;
+	connections: {
+		id: string;
+		mode: "write" | "read";
+	};
+}
+
+/**
+ * Token Provider implementation for connecting to an Azure Function endpoint for
+ * Azure Fluid Relay token resolution.
+ */
+export class AzureFunctionTokenProvider implements ITokenProvider {
+	/**
+	 * Creates a new instance using configuration parameters.
+	 * @param azFunctionUrl - URL to Azure Function endpoint
+	 * @param user - User object
+	 */
+	public constructor(
+		private readonly azFunctionUrl: string,
+		private readonly user?: Pick<AzureMember, "userId" | "userName" | "additionalDetails">,
+	) {}
+
+	public async fetchOrdererToken(tenantId: string, documentId?: string): Promise<ITokenResponse> {
+		return {
+			jwt: await this.getToken(tenantId, documentId),
```

Comment:
Nit
```suggestion
			jwt: await this.getToken(tenantId, documentId),
			fromCache: false,
```

Reply:
Since this is a reference implementation, I think providing the optional params explicitly is probably useful.